x/ccv/provider/keeper/keeper.go

```diff
@@ -621,6 +621,69 @@ func (k Keeper) DeleteInitChainHeight(ctx sdk.Context, chainID string) {
 	store.Delete(types.InitChainHeightKey(chainID))
 }
 
+// SetPendingVSCs sets in store the list of pending ValidatorSetChange packets under chain ID
+func (k Keeper) SetPendingVSCs(ctx sdk.Context, chainID string, packets []ccv.ValidatorSetChangePacketData) {
+	store := ctx.KVStore(k.storeKey)
+	var data [][]byte
+	for _, p := range packets {
+		pdata, err := p.Marshal()
+		if err != nil {
+			panic("failed to marshal ValidatorSetChange packet data")
+		}
+		data = append(data, pdata)
+	}
+	buf := &bytes.Buffer{}
+	err := json.NewEncoder(buf).Encode(data)
+	if err != nil {
+		panic("failed to encode json")
+	}
+	store.Set(types.PendingVSCsKey(chainID), buf.Bytes())
+}
+
+// GetPendingVSCs returns the list of pending ValidatorSetChange packets stored under chain ID
+func (k Keeper) GetPendingVSCs(ctx sdk.Context, chainID string) []ccv.ValidatorSetChangePacketData {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(types.PendingVSCsKey(chainID))
+	if bz == nil {
+		return nil
+	}
+	buf := bytes.NewBuffer(bz)
+
+	var data [][]byte
+	json.NewDecoder(buf).Decode(&data)
+
+	var packets []ccv.ValidatorSetChangePacketData
+	for _, pdata := range data {
+		var p ccv.ValidatorSetChangePacketData
+		err := p.Unmarshal(pdata)
+		if err != nil {
+			panic("failed to unmarshal ValidatorSetChange packet data")
+		}
+		packets = append(packets, p)
+	}
+
+	return packets
+}
+
+// AppendPendingVSC adds the given ValidatorSetChange packet to the list
+// of pending ValidatorSetChange packets stored under chain ID
+func (k Keeper) AppendPendingVSC(ctx sdk.Context, chainID string, packet ccv.ValidatorSetChangePacketData) {
+	packets := k.GetPendingVSCs(ctx, chainID)
+	packets = append(packets, packet)
+	k.SetPendingVSCs(ctx, chainID, packets)
+}
+
+// EmptyPendingVSC empties and returns the list of pending ValidatorSetChange packets for chain ID
```

Comment:
The comment doesn't accurately describe the functionality, because sometimes the list is deleted and sometimes it is found empty but not deleted. Can the comment be more precise?